app/controllers/projects_controller.rb

```diff
@@ -1,66 +1,72 @@
 class ProjectsController < ApplicationController
   before_action :authenticate_request, except: :index
   before_action :find_lesson
-  before_action :find_project, only: %i(update destroy)
+  before_action :find_project, only: %i(update destroy report)
 
   authorize_resource only: %i(update destroy)
 
   def index
-    @projects = projects
+    @projects = all_projects.page(params[:page]).per(50)
     @course = CourseDecorator.new(@lesson.course)
+    @user = current_user
+    @project = @projects.where(user_id: current_user.id).first if current_user
   end
 
   def create
     @project = new_project(project_params)
-    @project.save
-    set_recent_submissions
+    @project_saved = @project.save
+    @projects = latest_projects
   end
 
   def update
-    @project.update(project_params)
+    @project_updated = @project.update(project_params)
+    @decorated_project = ProjectDecorator.new(@project)
+  end
+
+  def report
+    @reported = ProjectReporter.report(
+      project: @project,
+      reporter: current_user,
+      description: params[:description]
+    )
   end
 
   def destroy
     @project.destroy
-    @project = new_project({})
-    set_recent_submissions
+    @project = new_project
+    @projects = latest_projects
   end
 
   private
 
-  def projects
-    Project
-      .all_submissions(@lesson.id)
-      .page(params[:page])
+  def all_projects
+    Project.all_submissions(@lesson.id).order(updated_at: :desc)
   end
 
-  def set_recent_submissions
-    @submissions = Project
-                   .all_submissions(@lesson.id)
-                   .where.not(user_id: current_user.id)
-                   .limit(10)
+  def new_project(parameters = {})
```

Comment:
Can we change this to use tap please?
```
current_user.projects.create!(parameters).tap do |project|
    project.lesson_id = @lesson.id
end
```

Reply:
I was initially using `tap` but found it harder to mock it in the tests. So, I resorted to the traditional way of doing. 

I'll try again to see if I can figure out how to mock it properly in the controller tests.